src/test/java/co/rsk/bitcoinj/script/FlyoverRedeemScriptParserTest.java

```diff
@@ -0,0 +1,291 @@
+
+    private final List<BtcECKey> defaultRedeemScriptKeys = RedeemScriptUtils.getDefaultRedeemScriptKeys();
+    private final List<BtcECKey> emergencyRedeemScriptKeys = RedeemScriptUtils.getEmergencyRedeemScriptKeys();
+    private final Sha256Hash derivationArgumentsHash = Sha256Hash.of(new byte[]{1});
+    private Script standardRedeemScript;
+    private Script flyoverStandardRedeemScript;
+    private Script erpRedeemScript;
+    private Script flyoverErpRedeemScript;
+    private Script p2shErpRedeemScript;
+    private Script flyoverP2shErpRedeemScript;
+
+    @Before
+    public void setUp() {
+        final long CSV_VALUE = 52_560L;
+        standardRedeemScript = RedeemScriptUtils.createStandardRedeemScript(defaultRedeemScriptKeys);
+        flyoverStandardRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), standardRedeemScript);
+
+        erpRedeemScript = RedeemScriptUtils.createErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), erpRedeemScript);
+
+        p2shErpRedeemScript = RedeemScriptUtils.createP2shErpRedeemScript(defaultRedeemScriptKeys, emergencyRedeemScriptKeys, CSV_VALUE);
+        flyoverP2shErpRedeemScript = RedeemScriptUtils.createFlyoverRedeemScript(derivationArgumentsHash.getBytes(), p2shErpRedeemScript);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsStandardRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverStandardRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsErpRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverErpRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getMultiSigType_whenIsP2shErpRedeemScript_shouldReturnFlyoverMultiSigType() {
+        // Arrange
+        FlyoverRedeemScriptParser flyoverRedeemScriptParser = new FlyoverRedeemScriptParser(flyoverP2shErpRedeemScript.getChunks());
+
+        // Act
+        MultiSigType actualMultiSigType = flyoverRedeemScriptParser.getMultiSigType();
+
+        // Assert
+        assertEquals(MultiSigType.FLYOVER, actualMultiSigType);
+    }
+
+    @Test
+    public void getM_whenFlyoverRedeemScriptContainsStandardRedeemScript_shouldReturnMValue() {
+        // Arrange
+        final int EXPECTED_M = 5;
```

Comment:
This can be moved to a class variable.

Reply:
Done! Good catch! This is a good way to get cohesion in the class.

Reply:
This method was refactored to apply a manual `ParameterizedTest`. Now, we have only one method needing this variable.